ocaml-lsp-server/src/compl.ml

```diff
@@ -79,42 +71,42 @@ let prefix_of_position ~short_path source position =
         | ':'
         | '~'
         | '#' ->
-          find (ch :: prefix) (i - 1)
-        | '.' ->
-          if short_path then
-            make_string prefix
-          else
-            find (ch :: prefix) (i - 1)
-        | _ -> make_string prefix
+          false
+        | '.' -> short_path
+        | _ -> true)
     in
-
-    let (`Offset index) = Msource.get_offset source position in
-    find [] (index - 1)
+    let pos =
+      match pos with
+      | None -> 0
+      | Some pos -> pos + 1
```

Comment:
is this why we stopped supporting non-prefix completions?

Reply:
You mean the non-prefix completion I've referred to in slack? No. I removed non-prefix completion because it was slow and unreliable.